SUSE/connect-ng
# collectors: add extra parameters for aarch64

The information we were gathering for aarch64 devices was incomplete. Now we will be passing also information that can be gathered from `device_tree` for non-ACPI compliant devices, or from `dmidecode` for ACPI compliant ones.

## How to test it

You need an ARM64 machine around (check inside of SUSE how you can get an ACPI compliant and a non-ACPI compliant one). With that, simply build SUSEConnect-ng on ARM64 (see `make build-arm` if you are on x86_64) and use the binary on an ARM64 machine with `--debug`. Expect:

On a non-ACPI compliant: a new `device_tree` element on the `hw_info` hash. Example on a Raspberry Pi 5:
```
{
  "hostname": "0e2318927679",
  "distro_target": "sle-15-aarch64",
  "hwinfo": {
    "arch": "aarch64",
    "container_runtime": "docker",
    "cpus": 4,
    "device_tree": "raspberrypi,5-model-bbrcm,bcm2712",   # <-- NOTE
    "hostname": "0e2318927679",
    "mem_total": 8052,
    "sockets": 1,
    "uuid": ""
  }
}
```

On an ACPI compliant device: a new `processor_information` hash with `family`, `manufacturer` and `signature`. Example on an Ampere Computing machine:

```
{
  "hostname": "ampere1",
  "distro_target": "sle-15-aarch64",
  "hwinfo": {
    "arch": "aarch64",
    "cpus": 32,
    "hostname": "ampere1",
    "mem_total": 256398,
    "processor_information": {   # <-- NOTE this new map
      "family": "ARMv8",
      "manufacturer": "AppliedMicro(R)",
      "signature": ""
    },
    "sockets": 1,
    "uuid": "4cfe0c5c-41c1-11e8-9f07-0cc47ad8d1fc"
  }
}
```
